jwst/datamodels/properties.py

```diff
@@ -56,8 +56,12 @@ def _as_fitsrec(val):
         return val
     else:
         coldefs = fits.ColDefs(val)
+        uint = any(c._pseudo_unsigned_ints for c in coldefs)
         fits_rec = fits.FITS_rec(val)
         fits_rec._coldefs = coldefs
+        # FITS_rec needs to know if it should be operating in pseudo-unsigned-ints mode,
+        # otherwise it won't properly convert integer columns with TZEROn before saving.
+        fits_rec._uint = uint
```

Comment:
Is this used in astropy.io.fits?

Reply:
Yes, [here](https://github.com/astropy/astropy/blob/de3933688433e22765b1bdffa175fcd954d691c3/astropy/io/fits/fitsrec.py#L940).  The else... is where it was changing the dtype to float64.